power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -569,6 +569,14 @@ template <symmetry_tag sym> struct NodeState {
     }
 };
 
+template <typename T> auto enumerate(T&& iterable) {
+    return std::views::transform(
+        std::views::iota(0, std::ranges::distance(iterable)),
+        [iterable = std::forward<T>(iterable)](auto i) -> std::pair<decltype(i), decltype(*std::begin(iterable))> {
+            return {i, *(std::begin(iterable) + i)};
+        });
+}
+
```

Comment:
This actually can not be used on clang and gcc ;(

Reply:
Will be cleaned up

Reply:
Do you know why is it disliked? Is it because it may allow attempting to forward rvalues as lvalues?

Reply:
gcc's support for ranges is meh

Reply:
Comment addressed.